oscore-cf/src/main/java/org/eclipse/californium/oscore/OSExceptionHandler.java

```diff
@@ -0,0 +1,158 @@
+	public static final String TYPE_NULL = "Type is null";
+	public static final String UNEXPECTED_OBSERVE = "An unexpected observe option";
+	public static final String SOURCE_ADRESS_NULL = "Source adress is null";
+	public static final String SOURCE_PORT_NULL = "Source port is null";
+	public static final String SOURCE_PORT_INVALID = "Source port is invalid";
+	public static final String ERROR_MESS_NULL = "Error message is null";
+	public static final String MID_INVALID = "MID is invalid";
+	public static final String MISSING_KID = "KID is missing";
+	public static final String EXCEPTION_NULL = "Exception is null";
+	public static final String REQUEST_NULL = "Request is null";
+	public static final String OPTIONSET_NULL = "OptionSet is null";
+	public static final String WRONG_VERSION_NBR = "Wrong version number";
+	public static final String BYTE_ARRAY_NULL = "Byte array is null";
+	public static final String NONCE_FAILED = "Nonce generation failed";
+	public static final String PARTIAL_IV_NULL = "PartialIV is null";
+	public static final String SENDER_ID_NULL = "SenderID is null";
+	public static final String COMMON_IV_NULL = "CommonIV is null";
+	public static final String NONCE_LENGTH_INVALID = "Nonce length is invalid";
+	public static final String COAP_CODE_INVALID = "Coap Code is invalid";
+
+	public static final String CANNOT_CREATE_ERROR_MESS = "Cannot create error message for this error";
+
+	/**
+	 * 
+	 * Process the exception e which was provoked by the received request.
+	 * 
+	 * @param e the exception which was thrown because of the received request.
+	 * @param request the received request.
+	 * @return the appropriate error message in a response.
+	 * @throws OSException
+	 */
+	public static Response manageError(OSException e, Request request) throws OSException {
+		String errMess = e.getMessage();
+		Response error = null;
+		boolean includeErrMess = true;
+
+		if(request == null){
+			LOGGER.severe(REQUEST_NULL);
+			return null;
+		}
+		
+		if (errMess != null) {
+			if (errMess.equals(CONTEXT_NOT_FOUND)) {
+				error = Response.createResponse(request, ResponseCode.UNAUTHORIZED);
+			} else if (errMess.equals(MISSING_KID)) {
+				error = Response.createResponse(request, ResponseCode.UNAUTHORIZED);
+			} else if (errMess.equals(DB_NULL)) {
+				error = Response.createResponse(request, ResponseCode.INTERNAL_SERVER_ERROR);
+				includeErrMess = false;
+			} else if (errMess.equals(FAILED_TO_DECODE_COSE)) {
+				error = Response.createResponse(request, ResponseCode.BAD_OPTION);
+			} else if (errMess.equals(DECRYPTION_FAILED)) {
+				error = Response.createResponse(request, ResponseCode.BAD_REQUEST);
+			} else if (errMess.equals(REPLAY_DETECT)) {
+				error = Response.createResponse(request, ResponseCode.SERVICE_UNAVAILABLE);
+				error.getOptions().setMaxAge(0);
+			} else {
+				LOGGER.severe(CANNOT_CREATE_ERROR_MESS);
+				return null;
+			}
+
+			Type tmp = Type.NON;
+			if (request != null) {
```

Comment:
request is already checked to be not null (line 81).

Reply:
Thanks.